packages/web-runtime/src/components/Topbar/NotificationsModal.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+    <oc-modal
+      :title="$gettext('Notification Settings')"
+      :button-cancel-text="$gettext('Discard')"
+      :button-confirm-text="$gettext('Save')"
+      @cancel="$emit('cancel')"
+      @confirm="onConfirm"
+    >
+      <template #content>
+        <oc-checkbox 
+          v-model="localNotifyEnabled" 
+          label="Disable all notifications" 
+        />
+      </template>
+    </oc-modal>
+  </template>
+  
+  <script lang="ts">
+  import { defineComponent, ref, watch, PropType } from 'vue'
+  import { useGraphClient } from 'web-pkg'
+  
+  export default defineComponent({
+    name: 'NotificationsModal',
+    emits: ['cancel', 'update:initialNotifyEnabled', 'confirm'],
+    props: {
+      initialNotifyEnabled: {
+        type: Boolean as PropType<boolean>,
+        required: true
+      },
+    },
+    setup(props, { emit }) {
+      const localNotifyEnabled = ref(props.initialNotifyEnabled);
+
+      watch(
+        () => props.initialNotifyEnabled,
+        (newVal) => {
+          localNotifyEnabled.value = newVal;
+        }
+      );
+
+      const onConfirm = () => {
+        emit('update:initialNotifyEnabled', localNotifyEnabled.value);
+        emit('confirm', localNotifyEnabled.value);
+      };
+
+      return {
+        localNotifyEnabled,
+        onConfirm,
+        ...useGraphClient()
+      };
+    }
+  })
+  </script>
+  
```

Comment:
You should lint this file to fix indentantion and remove some semicolons, I suggest you install the [volar vscode extension](https://marketplace.visualstudio.com/items?itemName=Vue.volar) and run its formatter.